packages/web-components/fast-foundation/src/number-field/number-field.ts

```diff
@@ -209,6 +209,7 @@ export class NumberField extends FormAssociatedNumberField {
         }
 
         if (value != this.value) {
+            this.displayText = value;
```

Comment:
This won't work. It breaks typing in values that aren't valid while you type them. Such as trying to type **-.5e10** isn't possible with this. We'd have to make adjustments to displayText in stepUp and stepDown methods.

We could update the displayText in **valueChanged()** to catch changes to value by script, but that could update it during the **updateValue()** method when we don't want it to.

One way to deal with it all is to create a flag, **updatingValue** that is set to true when **updateValue()** is called. It would set it to false after it is finished. Then we could just set **this.displayText** to **this.value** when **this.updatingValue** is equal to false. This would update when changed by script and would fix **stepUp** and **stepDown**.

Reply:
Cool...you want me to try and tackle this?

Reply:
I can get this. It would fix updating with scripts and increment and decrement. I'd like to build more tests around this to catch all of these edge cases as well.

Reply:
Also, just so there's no confusion, my change does not seem to inhibit the text entry you outlined.

Reply:
It was lagging and would sometimes replace the value with 0 when I was playing around with entering numbers.

Reply:
Ah, I see. Yeah, after playing with it a bit, the input does seem a little more fussy with my change present (not impossible, but definitely fussier). Still, I feel that the inevitable fix for [this Issue](https://github.com/microsoft/fast/issues/5249) might address that.

Reply:
I created a number-field with a min="10". With the new displayText property, it won't coerce the value to the min while typing. The internal value will actually be 10, but it will show what the user is typing now.